### DISCUSSION TITLE
OptionTemplate in ChoiceGroup as standard "list style"?

### DISCUSSION THREAD

**Discussion by RAN-Vestjylland** [BlazorFluentUI/BlazorFluentUI#328] [Q&A]
Hi all,

Is it possible to get a ChoiceGroup with the OptionTemplate (and own TItem type that is NOT inheriting IChoiceGroupOption) to look like a standard ChoiceGroup "list" as it's not using an OptionTemplate?

![2021-06-02 11_09_17-VES TBS3](https://user-images.githubusercontent.com/78351622/120453133-0cf62980-c393-11eb-92d7-70c24b7c3320.png)

Looking like the list below (I know it's not the same amount of option and text):
![2021-06-02 11_10_23-Blazor Fluent UI Webassembly Demo](https://user-images.githubusercontent.com/78351622/120453236-26977100-c393-11eb-83df-9fd04b8fad0a.png)

**Comment by RAN-Vestjylland** [comment]
How do you set so a choice is selected by default when the page is loaded?

**Reply by vnbaaij** [reply]
One way you could do this is for example in `ChoiceGroupPage.razor` is to add this to the code block:

`protected override void OnParametersSet()
  {
        _selectedCustomObject2 = _customObjects2[0];
        base.OnParametersSet();
  }`

**Reply by RAN-Vestjylland** [reply]
I wouldn't mind seeing an example of that on the test page if you don't mind. I don't seem to be able to make that work.
To me when this code is set:
`public ChoiceItem SelectedOwnRiskOption { get; set; }
`
`public IList<ChoiceItem> OwnRiskOptions { get; set; }
`
`<ChoiceGroup Label="Selvrisiko" ItemsSource="Model.OwnRiskOptions" TItem="ChoiceItem" Value="Model.SelectedOwnRiskOption" ValueChanged="OwnRiskChanged" />
`
When Model.SelectedOwnRiskOption gets set a value, it should set the selection, but it doesn't (in my code at least) *wink*wink*).

Hope you could clarify what I do wrong.

**Reply by vnbaaij** [reply]
It is in the ChoiceGroupPage now in the demo (https://www.blazorfluentui.net/choiceGroupPage) for the last example on the page. The first item of the group is selected. 

I didn't mess with the Value/Value Changed. Just kept the original bind-Value and set the value in OnParameterSet

**Reply by RAN-Vestjylland** [reply]
That is when the issues start. I need to be able to call an extra method on behalf of the choice that is happening in the ChoiceGroup. It will not allow me to have the ValueChanged when @bind-Value is set. The code complaints about ValueChanged can't be set twice. That is why I "only" have the Value attribute set together with TItem and ValueChanged. Also I'm not using the OptionTemplate where the @bind-SelectedOption is set. I don't want it to look like the big boxes like I mentioned earlier.
![2021-06-03 08_54_05-Greenshot](https://user-images.githubusercontent.com/78351622/120600733-66209480-c449-11eb-813a-301fefca4822.png)

**Reply by vnbaaij** [reply]
Yes, the complaint is logical. `@bind-Value` already creates `ValueChanged`. 
I created a gist that works on setting the initial value of the group and calls `ValueChanged` when other item is clicked: https://gist.github.com/vnbaaij/4b85d0076b20f1e0cb369b424d9ae07b

If you need the extra method to be called initially as well, then you would need to add that call to the `OnParameterSet`

**Reply by RAN-Vestjylland** [reply]
Ok got it working now. Thank you for the help.

**Comment by vnbaaij** [comment]
No, that is not possible as `ChoiceGroupOption.razor` either renders an `input` of `type=radio` (when `OptionTemplate` is null) or the `OptionTemplate` with some additional `<div>`s
